core/chains/constraints.go

```diff
@@ -0,0 +1,18 @@
+package chains
+
+// ID types represent unique identifiers within a particular chain type. Using string is recommended.
+type ID any
+
+// Config types should have fields for chain configuration, and normally implement sql.Scanner and driver.Valuer, but
+// that is not enforced here since legacy types used mixed pointer and value receivers.
+type Config interface {
+	// sql.Scanner
+	// driver.Valuer
+}
+
+// Node types should be a struct including these default fields:
```

Comment:
Agreed. Fortnately, the current uses are still tested and in the future with static config none of these will be necesssary - the DB fields (ID, created, updated) will go away, and the name likely doesn't have to be an internal field at all, since it's already a unique string to map on.

In the mean time, we can avoid cloning too much test code by adding a generic helper, or maybe finding some minimal set of properties to test for each one.